lib/Bindings/Python/circt/dialects/comb.py

```diff
@@ -4,3 +4,99 @@
 
 # Generated tablegen dialects end up in the mlir.dialects package for now.
 from mlir.dialects._comb_ops_gen import *
+
+from circt.support import NamedValueOpView
+
+from mlir.ir import IntegerAttr, IntegerType
+
+
+# Sugar classes for the various possible verions of ICmpOp.
+class ICmpOpBuilder(NamedValueOpView):
+
+  def operand_names(self):
+    return ["lhs", "rhs"]
+
+  def result_names(self):
+    return ["result"]
+
+  def __init__(self, predicate, data_type, input_port_mapping={}, **kwargs):
+    predicate = IntegerAttr.get(IntegerType.get_signless(64), predicate)
+    super().__init__(ICmpOp, data_type, input_port_mapping, [predicate],
+                     **kwargs)
+
+
+class EqOp:
```

Comment:
I made a decision here to add Python classes for the different variants of comparators. To me, the end user doing design entry doesn't care that we chose to represent all comparators as one op, they just want an equality comparator, or an unsigned less-than comparator. I'm not tied to this, but it is what I went for initially.

Some minor points about doing this:
* since these aren't defined in ODS, we can't add this sugar in the _ext module, so I've had to put them here
* I re-ordered where the "U" and "S" designators for signed and unsigned are relative to the ODS attributes, in order to make them match up with how we name other ops, like DivUOp and DivSOp

Reply:
I agree.